src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -283,32 +289,53 @@ protected override void CompleteAsyncMethodChecksBetweenStartAndFinish()
             }
         }
 
-        public override ImmutableArray<TypeParameterConstraintClause> GetTypeParameterConstraintClauses(bool canIgnoreNullableContext)
+        public override ImmutableArray<ImmutableArray<TypeWithAnnotations>> GetTypeParameterConstraintTypes()
         {
-            if (!_lazyTypeParameterConstraints.HasValue(canIgnoreNullableContext))
+            if (_lazyTypeParameterConstraintTypes.IsDefault)
             {
+                GetTypeParameterConstraintKinds();
```

Comment:
Is this necessary?

Reply:
> Is this necessary?

This function used to calculate both, the kinds and the types. I prefer to keep the same guarantee. Especially that we need to know both in order to resolve bounds. Also, it could be advantageous to reuse binders and syntax if we are going to perform full binding anyway.

---
In reply to: [520174573](https://github.com/dotnet/roslyn/pull/49227#discussion_r520174573) [](ancestors = 520174573)